src/ConnectedMode/Binding/PreSolutionBoundUpdateHandler.cs

```diff
@@ -0,0 +1,73 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core.Binding;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    [Export(typeof(PreSolutionBoundUpdateHandler))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class PreSolutionBoundUpdateHandler
+    {
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly IImportBeforeFileGenerator importBeforeFileGenerator;
+
+        private bool disposed;
+
+        [ImportingConstructor]
+        public PreSolutionBoundUpdateHandler(IActiveSolutionBoundTracker activeSolutionBoundTracker, IImportBeforeFileGenerator importBeforeFileGenerator)
+        {
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.importBeforeFileGenerator = importBeforeFileGenerator;
+
+            this.activeSolutionBoundTracker.PreSolutionBindingChanged += OnPreSolutionBindingChanged;
```

Comment:
I think my logic might be a bit flawed here, as the TriggerUpdate() will not be called when a user opens an already bound project. 

I am not sure this is the desired behaviour so please lmk

Reply:
> the TriggerUpdate() will not be called when a user opens an already bound project.

I thought it was called any time a solution is opened or closed, regardless of whether it is currently bound or not. Is it definitely not called when opening a bound solution?

Reply:
I have tried it a few times now and it seems that it does not get called

Reply:
Yea that seems to be the problem, I just saw this in the connectedMode package ->
```
// Trigger an initial update of suppressions (we might have missed the solution binding
// event from the ActiveSolutionBoundTracker)
// See https://github.com/SonarSource/sonarlint-visualstudio/issues/3886
var updater = componentModel.GetService<ISuppressionIssueStoreUpdater>();
updater.UpdateAllServerSuppressionsAsync().Forget();
```

but I dont think in our case that makes much sense, unless we do a check here -> "are we in connected mode" wdyt

Reply:
Yep. We could move the "is connected" check into the `TriggerUpdate` method of this class and then just have the package import the class and call `TriggerUpdate`. That would mean re-writing the `TriggerUpdate` method to look something like this:

```
public Task TriggerUpdateAsync()
{
    var config = configurationProvider.GetConfiguration();

    if (e.Configuration.Mode != SonarLintMode.Standalone)
    {
        await threadHandling.SwitchToBackGroundThread();

        // This should only happen once which is why it gets disposed after.
        importBeforeFileGenerator.WriteTargetsFileToDiskIfNotExists();
        Dispose();
    }
}
```

The simpler alternative would be to always install the targets, even when we are not in Connected Mode (which is what @rita-gorokhod originally suggested). If we did that then we could just have the package create and call `IImportBeforeFileGenerator` directly.

What do you think?

Reply:
I went with the first approach, simply because considering most of our users do not use connected made I feel like its better to not always put the files.